lib/capistrano/s3/publisher.rb

```diff
@@ -99,7 +99,10 @@ def self.base_file_path(root, file)
         end
 
         def self.files(deployment_path, exclusions)
-          Dir.glob("#{deployment_path}/**/*") - Dir.glob(exclusions.map { |e| "#{deployment_path}/#{e}" })
+          paths          = Dir.glob("#{deployment_path}/**/*", File::FNM_DOTMATCH)
```

Comment:
Not introduced by your changes, but we should `File.join` when building paths to ensure comptability with all OS (Windows)

Reply:
Refactored